src/grid/grid_api.F

```diff
@@ -331,4 +350,441 @@ SUBROUTINE get_rsgrid_properties(rsgrid, npts_global, npts_local, shift_local)
+   END SUBROUTINE grid_free_task_list
+
+! **************************************************************************************************
+!> \brief Collocate all tasks of in given list onto given grids.
+!> \param task_list ...
+!> \param ga_gb_function ...
+!> \param rs_grids ...
+!> \author Ole Schuett
+! **************************************************************************************************
+   SUBROUTINE grid_collocate_task_list(task_list, ga_gb_function, rs_grids)
+      TYPE(grid_task_list_type), INTENT(IN)              :: task_list
+      INTEGER, INTENT(IN)                                :: ga_gb_function
+      TYPE(realspace_grid_p_type), DIMENSION(:), POINTER :: rs_grids
+
+      CHARACTER(len=*), PARAMETER :: routineN = 'grid_collocate_task_list', &
+         routineP = moduleN//':'//routineN
+
+      INTEGER                                            :: ilevel, nlevels
+      INTEGER, ALLOCATABLE, DIMENSION(:), TARGET         :: border
+      INTEGER, ALLOCATABLE, DIMENSION(:, :), TARGET      :: npts_global, npts_local, shift_local
+      LOGICAL(KIND=C_BOOL)                               :: orthorhombic
+      LOGICAL(KIND=C_BOOL), ALLOCATABLE, DIMENSION(:), &
+         TARGET                                          :: distributed
+      REAL(KIND=dp), ALLOCATABLE, DIMENSION(:, :, :), &
+         TARGET                                          :: dh, dh_inv
+      REAL(KIND=dp), DIMENSION(:, :, :), POINTER         :: grid
+      TYPE(C_PTR), ALLOCATABLE, DIMENSION(:), TARGET     :: grid_pointers
+      TYPE(realspace_grid_type), POINTER                 :: rsgrid
+      INTERFACE
+         SUBROUTINE grid_collocate_task_list_c(task_list, orthorhombic, func, nlevels, &
+                                               npts_global, npts_local, shift_local, &
+                                               border, distributed, dh, dh_inv, grid) &
+            BIND(C, name="grid_collocate_task_list")
+            IMPORT :: C_PTR, C_INT, C_BOOL
+            TYPE(C_PTR), VALUE                        :: task_list
+            LOGICAL(KIND=C_BOOL), VALUE               :: orthorhombic
+            INTEGER(KIND=C_INT), VALUE                :: func
+            INTEGER(KIND=C_INT), VALUE                :: nlevels
+            TYPE(C_PTR), VALUE                        :: npts_global
+            TYPE(C_PTR), VALUE                        :: npts_local
+            TYPE(C_PTR), VALUE                        :: shift_local
+            TYPE(C_PTR), VALUE                        :: border
+            TYPE(C_PTR), VALUE                        :: distributed
+            TYPE(C_PTR), VALUE                        :: dh
+            TYPE(C_PTR), VALUE                        :: dh_inv
+            TYPE(C_PTR), VALUE                        :: grid
+         END SUBROUTINE grid_collocate_task_list_c
+      END INTERFACE
+
+      nlevels = SIZE(rs_grids)
+      CPASSERT(nlevels > 0)
+      orthorhombic = rs_grids(1)%rs_grid%desc%orthorhombic
```

Comment:
please wrap in `LOGICAL(..., C_BOOL)` to make it gcc-10 compatible

Reply:
Done via e31b98663872be7d523c84fcce4bd39b82cacb4a.